expr/literals.go

```diff
@@ -924,6 +946,42 @@ func LiteralFromProto(l *proto.Expression_Literal) Literal {
 				TypeParameters:   params,
 			},
 		}
+	case *proto.Expression_Literal_PrecisionTimestamp_:
+		precTimeStamp := lit.PrecisionTimestamp
+		precision, err := types.ProtoToTimePrecision(precTimeStamp.Precision)
+		if err != nil {
+			return nil
+		}
+		if precTimeStamp.Value < 0 {
+			return nil
+		}
+		return NewPrecisionTimestampLiteral(uint64(precTimeStamp.Value), precision, nullability)
+	case *proto.Expression_Literal_PrecisionTimestampTz:
+		precTimeStamp := lit.PrecisionTimestampTz
+		precision, err := types.ProtoToTimePrecision(precTimeStamp.Precision)
+		if err != nil {
+			return nil
+		}
+		if precTimeStamp.Value < 0 {
+			return nil
+		}
+		return NewPrecisionTimestampTzLiteral(uint64(precTimeStamp.Value), precision, nullability)
 	}
 	panic("unimplemented literal type")
 }
+
+func NewPrecisionTimestampLiteral(value uint64, precision types.TimePrecision, n types.Nullability) Literal {
```

Comment:
please add docs to this function and the one below.

Reply:
done